pkg/bus/meseeks.go

```diff
@@ -193,21 +276,79 @@ func (b *Bus) matchRequest(ctx context.Context, auth, userID, connID string,
 	// Set the actual UUID of the receiving user.
 	req.ReceivingUser.UserId = receiver.UUID
 
+	// Send new one or update here
+	exists, err := b.soughtGameStore.ExistsForUser(ctx, gameRequest.RequestId)
+	if err != nil {
+		return err
+	}
+
+	if exists {
+		return b.updateMatchRequest(ctx, auth, userID, connID, req)
+	} else {
+		return b.newMatchRequest(ctx, userID, receiver.UUID, connID, req)
+	}
+}
+
+func (b *Bus) newMatchRequest(ctx context.Context, userID string, receiverID string, connID string, req *pb.MatchRequest) error {
 	mg, err := gameplay.NewMatchRequest(ctx, b.soughtGameStore, req)
 	if err != nil {
 		return err
 	}
 	evt := entity.WrapEvent(mg.MatchRequest, pb.MessageType_MATCH_REQUEST)
 	log.Debug().Interface("evt", evt).Interface("req", req).Interface("receiver", mg.MatchRequest.ReceivingUser).
-		Str("sender", reqUser.UserId).Msg("publishing match request to user")
-	b.pubToUser(receiver.UUID, evt, "")
+		Str("sender", userID).Msg("publishing match request to user")
+	b.pubToUser(receiverID, evt, "")
 	// Publish it to the requester as well. This is so they can see it on
 	// their own screen and cancel it if they wish.
-	b.pubToConnectionID(connID, reqUser.UserId, evt)
+	b.pubToConnectionID(connID, userID, evt)
 
 	return nil
 }
 
+func (b *Bus) updateMatchRequest(ctx context.Context, auth string, userID string, connID string, req *pb.MatchRequest) error {
```

Comment:
This now might be called by the seeker or the receiver. Must make changes accordingly.